doc/specs/portable-mode-spec.md

```diff
@@ -0,0 +1,69 @@
+---
+author: Dustin L. Howett @DHowett
+created on: 2023-03-22
+last updated: 2023-03-22
+issue id: none
+---
+
+# Windows Terminal "Portable" Mode
+
+## Abstract
+
+Since we are planning on officially supporting unpackaged execution, I propose a special mode where Terminal stores its
+settings next to `WindowsTerminal.exe`.
+
+## Inspiration
+
+- [PortableApps](https://portableapps.com)
+- "Embeddable" Python, which relies on the deployment of a specific file to the Python root
+
+## Solution Design
+
+- _If running without package identity,_ `CascadiaSettings` will look for the presence of a file called `.portable` next
+  to `WindowsTerminal.exe`.
+- If that file is present, it will change the settings and state paths to be rooted at the directory containing
+  `WindowsTerminal.exe`.

```

Comment:
Should we maybe use a `settings` subdirectory? I'd not be too happy with cluttering the binary directory. Also, the existence of the `settings` directory could serve as an alternative indication for `.portable`.

Reply:
I agree with the subdirectory. I still want to add the marker file, though, just because _there's like a 50% likelihood that we add a `settings` directory ourselves_, and that I want switching portable on and off to be as easy as a single file switch. Rather than renaming `settings` to `settings-nah-not-today`

Reply:
I went with the subdirectory. Resolving! PR out in #15051